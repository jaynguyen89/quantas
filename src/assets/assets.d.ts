declare module '*.png' {
    const path: string;
    export default path;
}

declare module 'blob-polyfill';
