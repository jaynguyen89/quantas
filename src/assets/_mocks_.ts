import { IApiHotel, IHotel } from 'src/commons/interfaces';
import { CancellationType, ImageType, PromotionType, RatingType } from 'src/commons/enums';

export const mockedApiHotelData: {
    results: Array<IApiHotel>,
} = {
    results: [
        {
            id: '123',
            property: {
                propertyId: '111',
                title: 'First Hotel',
                address: ['1 What Street', 'Where 3333'],
                previewImage: {
                    url: 'https://photo1.png',
                    caption: 'Photo1',
                    imageType: 'PRIMARY',
                },
                rating: {
                    ratingValue: 3,
                    ratingType: 'self',
                },
            },
            offer: {
                promotion: {
                    title: 'Promo1',
                    type: 'MEMBER',
                },
                name: 'Promo1',
                displayPrice: {
                    amount: 111,
                    currency: 'AU',
                },
                savings: {
                    amount: 11,
                    currency: 'AU',
                },
                cancellationOption: {
                    cancellationType: 'NOT_REFUNDABLE',
                },
            },
        },
        {
            id: '456',
            property: {
                propertyId: '222',
                title: 'Second Hotel',
                address: ['2 Which Way', 'Anywhere 2222'],
                previewImage: {
                    url: 'https://photo2.png',
                    caption: 'Photo2',
                    imageType: 'PRIMARY',
                },
                rating: {
                    ratingValue: 4,
                    ratingType: 'star',
                },
            },
            offer: {
                promotion: {
                    title: 'Promo2',
                    type: 'MEMBER',
                },
                name: 'Promo2',
                displayPrice: {
                    amount: 222,
                    currency: 'US',
                },
                savings: null,
                cancellationOption: {
                    cancellationType: 'FREE_CANCELLATION',
                },
            },
        },
        {
            id: '789',
            property: {
                propertyId: '333',
                title: 'Third Hotel',
                address: ['3 Where Road', 'Somewhere 1111', 'Australia'],
                previewImage: {
                    url: 'https://photo3.png',
                    caption: 'Photo3',
                    imageType: 'PRIMARY',
                },
                rating: {
                    ratingValue: 5,
                    ratingType: 'self',
                },
            },
            offer: {
                promotion: {
                    title: 'Promo3',
                    type: 'MEMBER',
                },
                name: 'Promo3',
                displayPrice: {
                    amount: 333,
                    currency: 'VND',
                },
                savings: {
                    amount: 33,
                    currency: 'VND',
                },
                cancellationOption: {
                    cancellationType: 'FREE_CANCELLATION',
                },
            },
        },
    ],
};

export const mockedHotelData: Array<IHotel> = [
    {
        id: '123',
        name: 'First Hotel',
        address: '1 What Street, Where 3333',
        image: {
            url: 'https://photo1.png',
            caption: 'Photo1',
            type: ImageType.PRIMARY,
        },
        rating: {
            value: 3,
            type: RatingType.SELF,
        },
        pricing: {
            name: 'Promo1',
            price: {
                value: 111,
                currency: 'AU',
            },
            promotion: {
                title: 'Promo1',
                type: PromotionType.MEMBER,
            },
            savings: {
                value: 11,
                currency: 'AU',
            },
            cancellationType: CancellationType.FREE_CANCELLATION,
        },
    },
    {
        id: '456',
        name: 'Second Hotel',
        address: '2 Which Way, Where 2222',
        image: {
            url: 'https://photo2.png',
            caption: 'Photo2',
            type: ImageType.PRIMARY,
        },
        rating: {
            value: 4,
            type: RatingType.STAR,
        },
        pricing: {
            name: 'Promo2',
            price: {
                value: 222,
                currency: 'US',
            },
            promotion: {
                title: 'Promo2',
                type: PromotionType.MEMBER,
            },
            cancellationType: CancellationType.FREE_CANCELLATION,
        },
    },
    {
        id: '789',
        name: 'Third Hotel',
        address: '3 Where Road, Somewhere 1111, Australia',
        image: {
            url: 'https://photo3.png',
            caption: 'Photo3',
            type: ImageType.PRIMARY,
        },
        rating: {
            value: 5,
            type: RatingType.SELF,
        },
        pricing: {
            name: 'Promo3',
            price: {
                value: 333,
                currency: 'US',
            },
            promotion: {
                title: 'Promo3',
                type: PromotionType.MEMBER,
            },
            savings: {
                value: 33,
                currency: 'US',
            },
            cancellationType: CancellationType.FREE_CANCELLATION,
        },
    },
];