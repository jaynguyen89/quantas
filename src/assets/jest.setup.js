import { TextEncoder, TextDecoder, ReadableStream } from 'node:util';

global.TextEncoder = TextEncoder;
global.TextDecoder = TextDecoder;
global.ReadableStream = ReadableStream;