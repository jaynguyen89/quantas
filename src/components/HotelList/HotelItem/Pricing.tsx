import React from 'react';
import 'src/components/HotelList/HotelItem/hotelItem.scss';
import { IOffer } from 'src/commons/interfaces';

type TPricingProps = {
    pricing: IOffer,
};

const Pricing = ({
    pricing,
}: TPricingProps) => {
    return (
        <div className='pricing'>
            <p>1 night total ({pricing.price.currency})</p>
            <h5><span>$</span>{pricing.price.value}</h5>

            {pricing.savings && (
                <b>Save ${pricing.savings.value}<span>~</span></b>
            )}
        </div>
    );
};

export default Pricing;