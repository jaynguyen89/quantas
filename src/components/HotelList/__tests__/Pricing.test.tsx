import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import toJson from 'enzyme-to-json';
import { render } from '@testing-library/react';
import { mockedHotelData } from 'src/assets/_mocks_';
import Pricing from 'src/components/HotelList/HotelItem/Pricing';

configure({ adapter: new Adapter() });

describe('Pricing.tsx', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Pricing pricing={mockedHotelData[0].pricing} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('contains necessary elements', () => {
        const data = mockedHotelData[0].pricing;
        const wrapper = render(<Pricing pricing={data} />);

        const pricingContainer = wrapper.container.getElementsByClassName('pricing');
        expect(pricingContainer.length).toEqual(1);
        expect(pricingContainer[0]).toBeDefined();

        const pEl = pricingContainer[0].getElementsByTagName('p');
        expect(pEl.length).toEqual(1);
        expect(pEl[0]).toBeDefined();
        expect(pEl[0].innerHTML).toEqual(`1 night total (${data.price.currency})`);

        const h5El = pricingContainer[0].getElementsByTagName('h5');
        expect(h5El.length).toEqual(1);
        expect(h5El[0]).toBeDefined();
        expect(h5El[0].innerHTML).toContain(`<span>$</span>${data.price.value}`);

        const savingEl = pricingContainer[0].getElementsByTagName('b');
        expect(savingEl.length).toEqual(1);
        expect(savingEl[0]).toBeDefined();
        expect(savingEl[0].innerHTML).toContain(`Save $${data.savings!.value}<span>~</span>`);
    });

    it('does not contain the saving element', () => {
        const data = mockedHotelData[0].pricing;
        data.savings = undefined;
        const wrapper = render(<Pricing pricing={data} />);

        const pricingContainer = wrapper.container.getElementsByClassName('pricing');
        const savingEl = pricingContainer[0].getElementsByTagName('b');
        expect(savingEl.length).toEqual(0);
    });
});