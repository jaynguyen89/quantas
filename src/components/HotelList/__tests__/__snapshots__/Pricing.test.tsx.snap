// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing.tsx renders correctly 1`] = `
<div
  className="pricing"
>
  <p>
    1 night total (
    AU
    )
  </p>
  <h5>
    <span>
      $
    </span>
    111
  </h5>
  <b>
    Save $
    11
    <span>
      ~
    </span>
  </b>
</div>
`;
